controllers/restore/controller.go

```diff
@@ -67,6 +69,15 @@ func (r *RepoManagerRestoreReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, err
 	}
 
+	// if lock configmap is found it means that the restore already ran, so the controller should stop execution.
+	// To rerun a restore the user will have to manually delete the lock configmap first.
+	lockCM := &corev1.ConfigMap{}
+	if err := r.Get(ctx, types.NamespacedName{Name: CMLock, Namespace: pulpRestore.Namespace}, lockCM); err == nil {
+		controllers.CustomZapLogger().Warn("PulpRestore lock ConfigMap found. No restore procedure will be executed!")
+		controllers.CustomZapLogger().Warn("If you really want to run restore tasks again, just remove the " + CMLock + " ConfigMap")
```

Comment:
I don't know exactly what to say, but it would be good to give a warn about the DB issue,
"if you proceed, you may ... DB ..."
other than DB do we have more issues?
if so we can put a link to some documentation
```suggestion
		controllers.CustomZapLogger().Warn("If you really want to run restore tasks again, just remove the " + CMLock + " ConfigMap, For more info htttp://docs")
```

Reply:
> "other than DB do we have more issues?"

Yes, there is a risk (I'm saying risk because for object storage this would not be the case) of losing/overwriting all data created/added **after the restore execution**:
https://github.com/pulp/pulp-operator/blob/bd02fa9156bb9a0f1c0b51b3c9890b53fa6ba304/controllers/restore/pulp_dir.go#L29

which, from the restore controller point of view, is the right thing to do considering that we are asking it to restore the environment  ... unless we had kind of a "selective restoration".

Regarding referencing a doc, I think that we can work on it through https://github.com/pulp/pulp-operator/issues/765. WDYT?

Reply:
sounds good to me